overlord/hookstate/hookmgr.go

```diff
@@ -222,6 +222,25 @@ func hookSetup(task *state.Task) (*HookSetup, *snapstate.SnapState, error) {
 	return &hooksup, &snapst, nil
 }
 
+func undoHookSetup(task *state.Task) (*HookSetup, *snapstate.SnapState, error) {
+	var hooksup HookSetup
+	err := task.Get("undo-hook-setup", &hooksup)
+	if err != nil {
+		if err == state.ErrNoState {
```

Comment:
I suggest not having this here. The lack of symmetry often turns into bugs because we forget which is which. That means both of these functions are really a single one, with a task key name as a parameter. We can keep the individual functions calling the common underlying base function if that's convenient (depends on how many calls for these functions we have).

Reply:
Done